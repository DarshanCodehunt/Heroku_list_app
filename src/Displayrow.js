import React, { useState, useEffect } from 'react';


const Displayrow = ({ storyData, updateStory, fetchStoriesOntype }) => {
    const [type, setType] = useState('F');

    useEffect(() => {
        fetchStoriesOntype(type);
    }, [type])
    const fetchTime = (date) => {
        const d1 = new Date(new Date().toGMTString())
        const d2 = new Date(new Date(date).toGMTString())
        const diffTime = Math.abs(d1 - d2);
        const diffhours = Math.ceil(diffTime / (1000 * 60 * 60));
        return `${diffhours} hours ago [`;

    }

    const handleClick = (event, id) => {
        storyData.map((obj) => {
            if (obj.objectID === id) {
                obj.ishidden = true;
            }
        });

        updateStory(storyData);

    }

    const handleUpvote = (e, id) => {
        storyData.map((obj) => {
            if (obj.objectID === id) {
                obj.points = obj.points + 1;
                obj.upVoteChanged = true;
            }
        });
        updateStory(storyData);
    }

    const handlePagination = () => {
        type === 'F' ? setType('L') : setType('F');
    }
    const fetchRows = () => {
        let arr = storyData && storyData.map((obj, index) => {
            if (obj.ishidden) {
                return null;
            } else {
                let classNameDynamic = '';
                obj.upVoteChanged ? classNameDynamic = 'color-change' : classNameDynamic = '';
                return <div key={index} className='story-block'>
                    <p className='flex-item1'> {obj.num_comments}</p>
                    <p className='flex-item2'><span className={`upvote-label ${classNameDynamic}`}><i onClick={(e) => handleUpvote(e, obj.objectID)} class="fa fa-sort-asc" aria-hidden="true"></i>{obj.points}</span></p>
                    <p className='flex-item3'>{obj.title} <span>({obj.url} ) by</span> {obj.author} <span>{fetchTime(obj.created_at)}</span> <a onClick={(e) => handleClick(e, obj.objectID)}>Hide</a> <span>]</span></p>
                </div>
            }


        });
        if (arr) {
            arr.push(<div className='story-block'>
                <p className='flex-item1'> {''}</p>
                <p className='flex-item2'>{''}</p>
                <p onClick={handlePagination} className='flex-item3' style={{ color: 'orangered', cursor: 'pointer' }}>{type === 'F' ? 'More' : 'Back'}</p>
            </div>)
        }



        return arr;
    }

    return <div className='main-block'>
        {fetchRows()}

    </div>
}

export default Displayrow;