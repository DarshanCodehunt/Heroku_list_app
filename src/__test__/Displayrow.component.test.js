import React from 'react';
import { create } from 'react-test-renderer';
import Displayrow from '../Displayrow';


describe('<Displayrow> component', () => {
    test('Matches the snapshot', () => {
        const DisplayPage = create(
            <Displayrow />
        )
        expect(DisplayPage.toJSON()).toMatchSnapshot()
    })
})