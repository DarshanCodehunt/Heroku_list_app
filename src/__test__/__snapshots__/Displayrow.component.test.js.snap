// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Displayrow> component Matches the snapshot 1`] = `
<div
  className="main-block"
/>
`;
